Java/ds/AVL_Tree.java

```diff
@@ -0,0 +1,198 @@
+/*
+AVL tree is a self-balancing Binary Search Tree (BST) where the difference between heights of left and right subtrees cannot be more than one for all nodes and if at any time they differ by more than one, rebalancing is done to restore this property. Each node maintains extra information called a balance factor whose value is either -1, 0 or +1 where balance factor of a node in an AVL tree is the difference between the height of the left subtree and that of the right subtree of that node.
```

Comment:
Break this line into multiple smaller lines and justify the content. Even on a Desktop, its difficult to read